### ISSUE TITLE
`include_simulator_logs` in `scan` causes Exit status: 124

### ISSUE THREAD

**Issue by b-onc** [fastlane/fastlane#15205]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
When we include `include_simulator_logs: true` parameter in `scan`, the command
```
xcrun simctl spawn <UUID of simulator that ran the tests> log collect --output <path to output> 2>/dev/null
```
fails, and fastlane exists with status 124. When I run this command on my terminal, I get the output:
> An error was encountered processing the command (domain=com.apple.CoreSimulator.SimError, code=165):
> Invalid device state

because the device is `shutdown`. If I boot the device and run this command, I can get the logs.

##### Command executed
```
lane :test do
    cocoapods(try_repo_update_on_error: true)
    carthage
    scan(scheme: "MyAppTests",
             include_simulator_logs: true)
end
```

##### Complete output when running fastlane, including the stack trace and command used
I'm posting only the failing part:
```
...
[17:59:55]: ▸ 2019-08-19 17:59:55.278 xcodebuild[78053:1850548] [MT] IDETestOperationsObserverDebug: 103.553 elapsed -- Testing started completed.
[17:59:55]: ▸ 2019-08-19 17:59:55.281 xcodebuild[78053:1850548] [MT] IDETestOperationsObserverDebug: 0.004 sec, +0.004 sec -- start
[17:59:55]: ▸ 2019-08-19 17:59:55.281 xcodebuild[78053:1850548] [MT] IDETestOperationsObserverDebug: 103.553 sec, +103.549 sec -- end
[17:59:58]: ▸ Test Succeeded
+--------------------+-----+
|       Test Results       |
+--------------------+-----+
| Number of tests    | 135 |
| Number of failures | 0   |
+--------------------+-----+

[17:59:59]: ------------------------------
[17:59:59]: --- Collecting system logs ---
[17:59:59]: ------------------------------
[17:59:59]: $ xcrun simctl spawn ACC3C181-D5AA-4390-AE59-6B6D41568451 log collect --output /Users/user/app/fastlane/test_output/system_logs-iPhone\ 5s_iOS_12.2.logarchive 2>/dev/null

[17:59:59]: Exit status: 124
```

### Environment
<details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                         | Value                                                          |
| --------------------------- | -------------------------------------------------------------- |
| OS                          | 10.14.6                                                        |
| Ruby                        | 2.3.7                                                          |
| Bundler?                    | true                                                           |
| Git                         | git version 2.20.1 (Apple Git-117)                             |
| Installation Source         | /usr/local/bin/fastlane                                        |
| Host                        | Mac OS X 10.14.6 (18G87)                                       |
| Ruby Lib Dir                | /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib |
| OpenSSL Version             | LibreSSL 2.6.5                                                 |
| Is contained                | false                                                          |
| Is homebrew                 | false                                                          |
| Is installed via Fabric.app | false                                                          |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/                    |
| Xcode Version               | 10.2.1                                                         |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE |             |   |

**Comment by chrisballinger** [comment]
I am also seeing this on Xcode 11 GM seed 1 running on CircleCI.

```
+--------------------+----+
|      Test Results       |
+--------------------+----+
| Number of tests    | 12 |
| Number of failures | 0  |
+--------------------+----+

[13:42:28]: ------------------------------
[13:42:28]: --- Collecting system logs ---
[13:42:28]: ------------------------------
[13:42:28]: $ xcrun simctl spawn 11D27F33-48EC-41A8-875E-7D2446C12EBE log collect --output /Users/distiller/project/build/scan/system_logs-iPhone\ 5s_iOS_12.2.logarchive 2>/dev/null

[13:42:29]: Exit status: 124
```

**Comment by chrisballinger** [comment]
Even more strange - when running the lane locally `include_simulator_logs` is set to `false`, but when run on CircleCI `include_simulator_logs` is set to `true`. I don't see that behavior documented anywhere, so I'm not sure where it's coming from.

**Comment by b-onc** [comment]
@chrisballinger Are you using Xcode 11 on CircleCI? We're using Bitrise and it wasn't an issue with Xcode 10 on Bitrise, but after we've switched to Xcode11 it's also an issue there.

**Comment by OSSean** [comment]
I am seeing this on CircleCI with Xcode 11 GM Seed 2.

**Comment by chrisballinger** [comment]
Yep - Xcode 11 on CircleCI

**Comment by bpollman** [comment]
I’m seeing this as well. Found this post for xctool that worked around it by adding `--standalone` to that simctl call.

https://github.com/facebook/xctool/issues/766

**Comment by alexito4** [comment]
I'm also seeing this issue with Xcode 11 on CircleCI.

**Comment by pkcs12** [comment]
Having the same issue, not sure how to fix. Please help!

**Comment by ndonald2** [comment]
I found a fix in my setup. You need to explicitly pass `include_simulator_logs: false` when you invoke `scan` in your fastfile, as the option set in `Scanfile` seems to be ignored by fastlane in the Circle CI Xcode 11 image. For example:

```ruby
scan(include_simulator_logs: false)
```

This fixed the issue for me!

**Comment by marcomorain** [comment]
👋 I'm on the team at CircleCI that manages the macOS build system – I've just subscribed to this thread.

In the 1.0 build system we used to try to set as many env vars as we could to automatically store Fastlane logs as build artifacts with each build (this was handy for our users, but also for us to help debug builds that were failing mysterious ways). Specifically, we used to set `FL_BUILDLOG_PATH` to the artifacts directory.

On 2.0 we don't do any of this though as far as I know (it's been about 3 years since we wrote the 2.0 build system though, so my memory is hazy).

There might be code in Scan that detects "CI" or "CIRCLECI" being present as env vars and acts differently.

One thing I would like to see is _exactly_ what settings are being used by Scan. Is there a way to enable a verbose logging from scan that would print all settings?
